st2common/st2common/util/spec_loader.py

```diff
@@ -77,7 +77,7 @@ def load_spec(module_name, spec_file, allow_duplicate_keys=False):
 
     # 1. Check for duplicate keys
     if not allow_duplicate_keys:
-        yaml.load(spec_string, UniqueKeyLoader)
+        yaml.safe_load(spec_string)
```

Comment:
I missed this earlier - we intentionally use ``yaml.load()`` and not ``yaml.safe_load()`` here since we use a custom loader which detects duplicate keys which ``yaml.safe_load()`` doesn't support.

So using safe_load would change and break the existing behavior and that's not what we want.

Per my comment on other PR - this function is only used to load trusted openapi.yaml file which is controlled by us so not using ``safe_load()`` should be fine.

I suggest reverting this change + adding a comment to that line of code clarifying why we don't use safe_load there.

Reply:
EDIT: It looks like it's actually possible to achieve "detect duplicate keys" behavior with ``safe_load``, so we should do something like this instead - https://gist.github.com/danielpops/5a0726f2fb6288da749c4cd604276be8 (we of course can't just copy that code since it has no license which means we need to develop it ourselves or find similar snippet with an appropriate license).

Reply:
That gist looks super overcomplicated for what it's trying to achieve. I'm cooking a patch for PyYAML that should allow us to inherit from `SafeLoader` and customize it to disallow duplicate keys.

Reply:
@blag @Kami please let @ashwini-orchestral what she needs to do here. @blag if you are working on a patch, will you be including the change here in this PR?

Reply:
Two PRs to PyYAML to make this easier for us:
* yaml/pyyaml#502
* yaml/pyyaml#503

If one of those is accepted then it will allow this code to be much simpler.